azavea/district-builder-dtl-pa
# Add Staging deployment pipeline

## Overview

Adds a TravisCI Staging deployment pipeline. Cherry picks commits from:

- PublicMapping/DistrictBuilder#529
- PublicMapping/DistrictBuilder#540
- PublicMapping/DistrictBuilder#546
- PublicMapping/DistrictBuilder#548

Additionally, I modified Django settings to set multiple `ALLOWED_HOSTS` so that we can access the app from either the CloudFront URL, or the App Server origin URL.

Finally, I removed positional arguments from `scripts/load_configured_data` and added  `-p/--production` and `-c/--config` flags to avoid confusion when running in production. Previously, the script expected the second argument to be the `configPath`, which breaks when the script is run with just a `--production` flag.

### Checklist

- [ x ] PR has a name that won't get you publicly shamed for vagueness

### Demo

https://pa.districtbuilder.azavea.com

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.


## Testing Instructions
* The deployment instructions were tested in PublicMapping/DistrictBuilder#546; I don't think they need to be revisited.
* I deployed this branch to staging from Travis. See [build output](https://travis-ci.org/azavea/district-builder-dtl-pa/builds/386337727) 
* Configure Docker to communicate with the App Server daemon over TLS, and run `scripts/load_configured_data --production`.

```bash
$ aws --profile district-builder s3 cp s3:/districtbuilder-staging-config-us-east-1/docker_certs/client/client.zip .
$ mkdir -p /tmp/districtbuilder-docker-certs
$ unzip -d /tmp/districtbuilder-docker-certs client.zip
$ export DOCKER_HOST=origin.pa.districtbuilder.azavea.com:2476
$ export DOCKER_TLS_VERIFY=1
$ export DOCKER_CERT_PATH=/tmp/districtbuilder-docker-certs
$ export  GIT_COMMIT=efe7d01
$ mv .env .env.dev
$ aws --profile district-builder s3 cp s3://districtbuilder-staging-config-us-east-1/terraform/pa/.env .
$ export DB_DEBUG=1
$ ./scripts/load_configured_data --production
```

Closes #19
